src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/nestedfield/Where.java

```diff
@@ -67,8 +68,9 @@ private void useAnyChildTag(Scope scope) {
      * Merge anyway if the root of WHERE clause be reached
      */
     private void mergeIfHaveTagAndIsRootOfWhere(Scope scope) {
-        if (!scope.getConditionTag(expr).isEmpty()
-                && expr.getParent() instanceof MySqlSelectQueryBlock) {
+        if ((!scope.getConditionTag(expr).isEmpty()
+             && expr.getParent() instanceof MySqlSelectQueryBlock)
+            || expr.getParent() instanceof SQLNotExpr) {
```

Comment:
This line makes the method body inconsistent with its name.

Reply:
Done.